src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -1,148 +1,297 @@
 package htsjdk.samtools.cram.build;
 
 import htsjdk.HtsjdkTest;
+import htsjdk.samtools.SAMRecord;
 import htsjdk.samtools.cram.ref.ReferenceContext;
-import htsjdk.samtools.cram.structure.Container;
-import htsjdk.samtools.cram.structure.CramCompressionRecord;
-import htsjdk.samtools.cram.structure.CRAMStructureTestUtil;
-import htsjdk.samtools.cram.structure.Slice;
+import htsjdk.samtools.cram.structure.*;
 import org.testng.Assert;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.sql.Ref;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-/**
- * Created by vadim on 15/12/2015.
- */
 public class ContainerFactoryTest extends HtsjdkTest {
-    private static final int TEST_RECORD_COUNT = 10;
-    private static final int READ_LENGTH_FOR_TEST_RECORDS = CRAMStructureTestUtil.READ_LENGTH_FOR_TEST_RECORDS;
 
-    @Test
-    public void recordsPerSliceTest() {
-        final int recordsPerSlice = 100;
-        final ContainerFactory factory = new ContainerFactory(CRAMStructureTestUtil.getSAMFileHeaderForTests(), recordsPerSlice);
-
-        // build a container with the max records per slice
-
-        final List<CramCompressionRecord> records = CRAMStructureTestUtil.getSingleRefRecords(recordsPerSlice, 0);
-        final long dummyByteOffset = 0;
-        final Container container = factory.buildContainer(records, dummyByteOffset);
+    @DataProvider(name="singleContainerSliceDistribution")
+    private Object[][] getSingleContainerSliceDistribution() {
+        final int RECORDS_PER_SLICE = 100;
```

Comment:
You don't seem to have any test cases that vary the number of records per slice -- would it be meaningful to test with different values of that parameter?

Reply:
I don't think so (and there are quite a few tests in other places that do vary that value). In this case its deliberately using an artificially small number as a starting basis for testing around the boundary/state transitions.